internal/engine/experiment/riseupvpn/riseupvpn.go

```diff
@@ -198,9 +208,10 @@ func (m Measurer) Run(ctx context.Context, sess model.ExperimentSession,
 		tk := entry.TestKeys
 		testkeys.AddCACertFetchTestKeys(tk)
 		if tk.Failure != nil {
+			// TODO(bassosimone,cyberta): should we update the testkeys
+			// in this case (e.g., APIFailure?)
```

Comment:
@cyBerta what do you think?

Reply:
If the served CA cert is invalid but a connection to the endpoint can be made, it is a sign for a misconfigured provider (expired cert). Adding a new error api_misconfigured might be interesting here as well. If the connection was blocked  APIFailure = true would make sense though.

Reply:
see my comment https://github.com/ooni/probe-cli/pull/275#discussion_r606182668